File: library/src/main/java/com/ebr163/inspector/rule/RegexpRule.java
package com.ebr163.inspector.rule;

import java.util.regex.Pattern;

/**
 * Created by Bakht
 * on 11.01.2018.
 */

public class RegexpRule extends AbstractRule<String> {

    private final String regexpPattern;

    public RegexpRule(String regexpPattern, String errorMessage) {
        super(errorMessage);
        this.regexpPattern = regexpPattern;
    }

    @Override
    public boolean verify(String value) {
        return Pattern.compile(regexpPattern).matcher(value).matches();
    }
}

File: library/src/main/java/com/ebr163/inspector/rule/RegexpRule.kt
package com.ebr163.inspector.rule

import java.util.regex.Pattern

/**
 * Created by Bakht
 * on 11.01.2018.
 */

class RegexpRule(private val regexpPattern: String, errorMessage: String) : AbstractRule<String>(errorMessage) {

    override fun verify(value: String?): Boolean = Pattern.compile(regexpPattern).matcher(value!!).matches()
}
